src/figuresClasses.ts

```diff
@@ -1,19 +1,97 @@
-export interface Figure {
+type ShapeType = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: ShapeType;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: ShapeType = 'triangle';
```

Comment:
and here you can use 'values by enum keys')

Reply:
agree